api/src/backend/api/v1/serializers.py

```diff
@@ -1308,12 +1308,13 @@ class Meta:
             "inserted_at": {"read_only": True},
             "updated_at": {"read_only": True},
             "provider": {"read_only": True},
-            "secret_type": {"read_only": True},
         }
 
     def validate(self, attrs):
         provider = self.instance.provider
-        secret_type = self.instance.secret_type
+        # To allow updating a secret with the same type without making the `secret_type` mandatory
+        # TODO: review API spec
```

Comment:
@AdriiiPRodri @vicferpoy I'm not sure if this requires regenerating API spec. Moreover, since the current one has commented code I'm not being able to clearly see differences.